web/src/stream_settings_ui.js

```diff
@@ -502,7 +502,10 @@ export function switch_stream_tab(tab_name) {
     }
 
     redraw_left_panel();
-    stream_edit.setup_subscriptions_tab_hash(tab_name);
+
+    if (!maybe_reset_right_panel()) {
+        stream_edit.setup_subscriptions_tab_hash(tab_name);
```

Comment:
I think there can be a better way to implement this. `maybe_reset_right_panel` can be updated to
```js
export function maybe_reset_right_panel() {
    if ($(".stream-row.active").hasClass("notdisplayed")) {
        stream_edit.empty_right_panel();
    }
}
```

And then just call `maybe_reset_right_panel` and `setup_subscriptions_tab_hash` here without any if blocks, since the hash is changed only if required in `setup_subscriptions_tab_hash`.

Reply:
Maybe we can remove `maybe_reset_right_panel` as well and just have the code directly here since we do not re-use it.